### PR TITLE
DAYS DUE BRANCH 4.4:

### PR DESCRIPTION
**Pull request by menechel**
- Modified TaskDaysDueElementWdg to add 2 and 1 hour messages and colour changes and improved the messages and measuring differences between dates
- Changed get dir info to look at 3 ### instead of 1 #

### CODE REVIEW

**Comment by menechel** [comment]
@oboreo can you please review?

File: src/tactic/ui/table/statistic_wdg.py [hunk 1]
```diff
@@ -518,13 +518,24 @@ def init_data(my):
...
         from pyasm.common import SPTDate
-        today = SPTDate.start_of_today()
+        import datetime
+        import time
+        today = datetime.datetime.now()
 
         # get the difference
         delta = due_date - today
-        diff = delta.days
+        diff = (delta.days * 86400) + delta.seconds
```
Comment (oboreo) [inline src/tactic/ui/table/statistic_wdg.py]: should define a class constant called DAY = 86400, HOUR = 3600... so you can do DAY * 2 later or HOUR *2

File: src/tactic/ui/table/statistic_wdg.py [hunk 2]
```diff
@@ -518,13 +518,24 @@ def init_data(my):
 
         due_date = parser.parse(value)
 
+        import calendar
+        from datetime import datetime, timedelta
+
+        # convert the due date from UTC to local time
+        timestamp = calendar.timegm(due_date.timetuple())
+        local_dt = datetime.fromtimestamp(timestamp)
+        assert due_date.resolution >= timedelta(microseconds=1)
```
Comment (oboreo) [inline src/tactic/ui/table/statistic_wdg.py]: what is this assert error for? In general , don't put assert error, tactic can't handle this type of error well...

File: src/tactic/ui/table/statistic_wdg.py [hunk 3]
```diff
@@ -518,13 +518,24 @@ def init_data(my):
...
+        due_date = local_dt.replace(microsecond=due_date.microsecond)
+
         # get today's date
         from pyasm.common import SPTDate
-        today = SPTDate.start_of_today()
+        import datetime
+        import time
+        today = datetime.datetime.now()
```
Comment (oboreo) [inline src/tactic/ui/table/statistic_wdg.py]: rename this variable as now

**Pull request merged**